feat: integrate Dojo hooks in Aquariums page

Sandijigs opened the pull request:
- Replace mock data with real player aquariums using getPlayerAquariums()
- Implement aquarium loading with fish using Promise.all() for parallel fetching
- Add real aquarium creation using newAquarium() contract function
- Connect aquarium selection with complete data loading before navigation
- Update dynamic statistics to use real contract data
- Add comprehensive error handling and loading states
- Transform contract data models to UI-compatible format
- Add wallet connection validation before contract calls

## 🔥 Pull Request: [Descriptive Title]

### 📌 Related Issue  
Closes #227 

### 📝 Description  
  This PR integrates Dojo hooks in the Aquariums page to replace mock data with real contract data. The implementation connects the frontend to the deployed Dojo contracts on Sepolia testnet, enabling real player aquarium management, fish loading, and aquarium creation through blockchain transactions.

### ✅ Changes Made  
- [ ] Backend  
- [x] Frontend   

### 📷 Evidence  
- **Frontend:** Attach a screenshot.  
<img width="1680" height="962" alt="Screenshot 2025-08-17 at 9 18 00 pm" src="https://github.com/user-attachments/assets/06748c9d-dc0b-4ad2-bc9a-8654bc916317" />

- **Backend:** Attach an image of the test execution results.

### 🚀 Additional Notes  
_Any extra comments about this PR (if applicable)._  


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Aquariums and fish now load live from your connected wallet.
  * Create a new on-chain aquarium directly from the page (wallet required).
  * Selecting an aquarium preloads its fishes before entering the game.

* **Enhancements**
  * Loading indicators and clear error banners for fetch errors.
  * Stats reflect real-time totals, averages, and premium counts.
  * Contextual messaging when you have no aquariums.
  * Automatic refresh when your wallet/account changes.

* **Chores**
  * Added Prettier (dev) and updated ignore rules for lockfiles and build outputs.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Sandijigs committed:
feat: integrate Dojo hooks in Aquariums page

- Replace mock data with real player aquariums using getPlayerAquariums()
- Implement aquarium loading with fish using Promise.all() for parallel fetching
- Add real aquarium creation using newAquarium() contract function
- Connect aquarium selection with complete data loading before navigation
- Update dynamic statistics to use real contract data
- Add comprehensive error handling and loading states
- Transform contract data models to UI-compatible format
- Add wallet connection validation before contract calls

Sandijigs commented:
@Josue19-08 . help review my the PR i just made

KevinMB0220 commented:
@Sandijigs please fix the code format in the frontend

Josue19-08 commented:
@Sandijigs, make sure your frontend CI is working properly. Check out Prettier or ESLint.

```
Run pnpm format:check

> Aqua-Stark@0.0.0 format:check /home/runner/work/Aqua-Stark/Aqua-Stark/client
> prettier --check .

Checking formatting...
[warn] src/pages/aquariums.tsx
[warn] Code style issues found in the above file. Run Prettier with --write to fix.
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

Sandijigs committed:
Merge branch 'main' into Integrate-Dojo-Hooks

KevinMB0220 commented:
@Sandijigs please format the code you do it, because the ci fail for that

Sandijigs committed:
fix: resolve formatting issues and add prettier dev dependency

- Fix code formatting in src/pages/aquariums.tsx
- Add prettier as dev dependency in package.json
- Update pnpm-lock.yaml with prettier dependency
- Ensure all files pass prettier format check

Josue19-08 commented:
@Sandijigs Make sure to correct the formatting
```
Checking formatting...
[warn] package-lock.json
[warn] Code style issues found in the above file. Run Prettier with --write to fix.
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

Josue19-08 commented:
@coderabbitai
Very briefly, is this issue complete? #227

Josue19-08 commented:
> @Sandijigs Make sure to correct the formatting
> 
> ```
> Checking formatting...
> [warn] package-lock.json
> [warn] Code style issues found in the above file. Run Prettier with --write to fix.
>  ELIFECYCLE  Command failed with exit code 1.
> Error: Process completed with exit code 1.
> ```

@Sandijigs apply this formatting and I'll merge

Sandijigs committed:
fix CI issues
